workflows/nanopore.nf

```diff
@@ -22,16 +22,68 @@ if (params.input) { ch_input = file(params.input) } else { exit 1, 'Input sample
     IMPORT NF-CORE MODULES
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
+include { PORECHOP_ABI                           } from '../modules/nf-core/porechop/abi/main'
+include { FLYE                                   } from '../modules/nf-core/flye/main'
+include { RACON                                  } from '../modules/nf-core/racon/main'
+include { CUSTOM_DUMPSOFTWAREVERSIONS            } from '../modules/nf-core/custom/dumpsoftwareversions/main'
 
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     IMPORT LOCAL MODULES/SUBWORKFLOWS
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
+include { INPUT_CHECK                            } from '../subworkflows/local/input_check'
+include { MINIMAP2_SUBWORKFLOW                   } from '../subworkflows/local/minimap2_subworkflow'
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     RUN MAIN WORKFLOW
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
+
+workflow NANOPORE {
```

Comment:
Just for my noob self: why is the nanopore pipeline quite different from the illumina one? what are the main differences?

Reply:
Once you have an assembly and mapped reads (BAM or SAM files) the downstream analysis/program steps are the same. Getting to that point for short and long reads is different because the assembly/alignment algorithms are very different for short vs long reads, both because of the length and because of how error-prone long reads tend to be (although this is rapidly changing and getting better). The similar steps are QC/adapter removal, assembly, and mapping - the tools just differ for prepping short and long reads. Which is why they are in separate workflows like this. There are also QC tools for long reads where you can filter by length so you aren't including short reads which for long read specific assemblers can mess with things. Long read prep also has the additional steps of polishing, which previously before the updated Nanopore kits it would be very typical to polish 3 separate times with racon, 3 with medaka and then polishing with the corresponding short reads (if you had them). But now that the adapter ligation kits for Nanopore are better and the accuracy is better, polishing with corresponding short reads makes things worse in highly repetitive areas like rRNA, so I won't add that capability here. 
It's also been pretty popular recently to do hybrid assembly of short and long reads (metaspades can do this for example) but it creates really fragmented assemblies in my experience, especially if you have a lot strain heterogeneity. The nf-core/mag pipeline does this for example - it doesn't treat short and long reads separately. However for our use cases I'm fairly positive if we had one or the other we would be treating them separately because of how the technology has gotten better for long reads.